test/aleph/classpath_test.clj

```diff
@@ -0,0 +1,42 @@
+(ns aleph.classpath-test
+  (:require  [clojure.test      :refer [deftest testing is]]
+             [aleph.http        :as http]
+             [manifold.deferred :as d]
+             [signal.handler    :refer [with-handler]])
+  (:import [io.netty.util.concurrent Future]
+           [java.lang.management ManagementFactory]
+           [java.util.concurrent CompletableFuture]))
+
+(defn- operation-complete
+  "Stubs for `GenericFutureListener/operationComplete` which
+  returns a completed `CompletableFuture` containing either
+  `true` or `false` in case of Throwable."
+  [^CompletableFuture result ^Future f d]
+  (try (d/success! d (.getNow f))
+       (.complete result true)
+       d
+       (catch Throwable _ (do (.complete result false) d))))
+
+(defn pid
+  "Gets this process' PID."
+  []
+  (let [pid (.getName (ManagementFactory/getRuntimeMXBean))]
+    (->> pid
+         (re-seq #"[0-9]+")
+         (first)
+         (Integer/parseInt))))
```

Comment:
Does this work in all JVMs? I see for a JVM 10+ at least, there's a `.getPid()` method that might be better. Maybe we can use that if present, and fall back to this if not?

Reply:
I'm using `ProcessHandler` when available now.